### PR TITLE
feat: add cert-manager

### PR DESCRIPTION
**Pull request by ghost**
This installs cert-manager in the playground for testing purposes.

### CODE REVIEW

File: Provisioning/Vault/cert-manager-demo.tf [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# This is an example for a demo app with Ingress for demonstrating cert-manager
+# + Vault PKI/ACME capabilities
+
+locals {
+  manifests = fileset("./manifests/cert-manager-demo", "*.yaml")
```
Comment (in0rdr) [inline Provisioning/Vault/cert-manager-demo.tf:5]: added all yaml manifests for this example in a folder

File: Provisioning/Vault/audit.tf [hunk 2]
```diff
@@ -0,0 +1,7 @@
+resource "vault_audit" "file" {
+  type = "file"
+
+  options = {
+    file_path = "stdout"
```
Comment (in0rdr) [inline Provisioning/Vault/audit.tf:5]: wanted to see the certificates issued so I enabled the audit log to container stdout

File: Provisioning/Vault/cert-manager-demo.tf [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+resource "vault_generic_endpoint" "config-acme" {
+  path                 = "${vault_mount.pki.path}/config/cluster"
+  ignore_absent_fields = true
+  disable_delete       = true
+  write_fields         = ["path", "aia_path"]
+  data_json            = <<EOT
+{
+  "path": "https://vault.vault.svc.cluster.local:8200/v1/${vault_mount.pki.path}",
```
Comment (in0rdr) [inline Provisioning/Vault/cert-manager-demo.tf:33]: it is crucial that these paths correspond to the server in the cert-manager cluster issuer

File: Provisioning/Vault/cert-manager-demo.tf [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  backend = vault_mount.pki.path
+  issuing_certificates = [
+    "{{cluster_aia_path}}/issuer/{{issuer_id}}/pem",
+  ]
+  crl_distribution_points = [
+    "{{cluster_aia_path}}/issuer/{{issuer_id}}/crl/pem",
+  ]
+}
+
+resource "vault_generic_endpoint" "acme-headers" {
```
Comment (in0rdr) [inline Provisioning/Vault/cert-manager-demo.tf:65]: these also includes some recommendations regarding auditing, check `vault pki health-check pki` for details

File: Provisioning/Vault/manifests/cert-manager-demo/ingress.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    # - https://cert-manager.io/docs/concepts/certificate/#certificate-lifecycle
+    # - https://cert-manager.io/docs/troubleshooting/acme/#http01-troubleshooting
+    # - https://cert-manager.io/docs/usage/ingress/#supported-annotations
+    # There may be an issue with your ingress controller handling multiple
+    # resources for the same hostname
+    #
+    # NAME                        CLASS   HOSTS         ADDRESS         PORTS     AGE
+    # cert-manager-demo           nginx   example.com   192.168.39.94   80, 443   4m21s
+    # cm-acme-http-solver-xrt22   nginx   example.com   192.168.39.94   80        84s
+    acme.cert-manager.io/http01-edit-in-place: "true"
```
Comment (in0rdr) [inline Provisioning/Vault/manifests/cert-manager-demo/ingress.yaml:21]: really weird, but after hours of debugging I figured it out :)

File: Vault-Deployment/add-to-hosts.sh [hunk 6]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
Comment (in0rdr) [inline Vault-Deployment/add-to-hosts.sh:1]: small bug, not related to the PR here

File: Vault-Deployment/values/vault-values.yaml [hunk 7]
```diff
@@ -7,7 +7,7 @@ injector:
 server:
   image:
     repository: hashicorp/vault-enterprise
-    tag: 1.13.1-ent
+    tag: 1.15.2-ent
```
Comment (in0rdr) [inline Vault-Deployment/values/vault-values.yaml:10]: acme was introduced in Vault >=1.14

**Review by Xelef2000 — APPROVED** [review]

**Pull request merged**